cmake/FileUtil.cmake

```
@@ -141,6 +141,20 @@ function(ly_file_read path content)
     set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${path})
 endfunction()
 
+#! ly_file_read: wrap ly_file_read but store the file in a cache to avoid 
+#  extra reads.
+function(ly_file_read_cache path content)
```

maybe we should start naming functions starting with `o3de` now?
```suggestion
function(o3de_file_read_cache path content)
```